src/main/java/org/jenkinsci/test/acceptance/po/Job.java

```diff
@@ -444,8 +445,10 @@ public ScmPolling pollScm() {
      */
     public void delete() {
         open();
-        clickLink("Delete Project");
+        WebElement deleteLink = find(by.link("Delete Project"));
+        deleteLink.click();
         confirmAlert(2);
+        waitFor(deleteLink).until(Control::isStale);
```

Comment:
Same thing, causes a full reload page.
Wait until the "delete" button is stale. The webElement must be obtained before confirming the alert; if try to obtain it after, the element may not be already in the DOM.